simulator/src/data/undo.js

```diff
@@ -17,9 +17,9 @@ import { scheduleBackup } from './backupCircuit';
  * @exports undo
  */
 export default function undo(scope = globalScope) {
-    if (layoutModeGet()) return;
-    if (scope.backups.length === 0) return;
     scheduleBackup(); // Backup current globalscope
+    if (layoutModeGet()) return;
+    if (scope.backups.length < 2) return;
```

Comment:
I think this condition is good enough, we don't need to check `backUp(scope).allNodes.length !== 0`. What do you say?